src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -373,36 +375,30 @@ protected Blob fetchByPath(String tenantId, @Nonnull String path) {
         }
     }
 
-    protected String ensureRelativePath(String path) {
-        if (path.startsWith("/")) {
-            return path.substring(1);
-        }
-
-        return path;
-    }
-
     @Override
     public Optional<? extends Blob> findByPath(String path) {
         return findByPath(UserContext.getCurrentUser().getTenantId(), path);
     }
 
     @Override
     public Blob findOrCreateByPath(String tenantId, String path) {
-        if (Strings.isEmpty(path)) {
+        String sanitizedPath = utils.sanitizePath(path);
+
+        if (Strings.isEmpty(sanitizedPath)) {
             throw new IllegalArgumentException("An empty path was provided!");
         }
 
-        String key = determinePathCacheKey(tenantId, path);
+        String key = determinePathCacheKey(tenantId, sanitizedPath);
         Blob blob = blobByPathCache.get(key);
         if (blob == null) {
-            blobByPathCache.remove(key);
```

Comment:
wasn't this put here intentionally as removed is distributed over the intercom and put isn't? @idlira

Reply:
Yup, put is not distributed